drivers/include/lm75.h

```diff
@@ -0,0 +1,224 @@
+    uint8_t     os_shift;    /**< how many bits need to be shifted (2 bytes - any unused bits) */
+    uint8_t     temp_shift;  /**< how many bits need to be shifted (2 bytes - any unused bits) */
+} lm75_temp_resolutions_t;
+
+/**
+ * @brief lm75a temperature resolutions
+ */
+static const lm75_temp_resolutions_t tmp1075_res = {
+    .os_res = 625,           /**< resolution in 0.0625ºC */
+    .os_mult = 10000,        /**< Must multiply by 10000 to get temp in ºC */
+    .os_shift = 4,           /**< Only the 12 most significant bits are needed */
+    .temp_res = 625,         /**< resolution in 0.0625ºC */
+    .temp_mult = 10000,      /**< Must multiply by 10000 to get temp in ºC */
+    .temp_shift = 4,         /**< Only the 12 most significant bits are needed */
+};
+
+/**
+ * @brief tmp1075 temperature resolutions
+ */
+static const lm75_temp_resolutions_t lm75a_res = {
+    .os_res = 5,        /**< resolution in 0.5ºC */
+    .os_mult = 10,      /**< Must multiply by 10 to get temp in ºC */
+    .os_shift = 7,      /**< Only the 9 most significant bits are needed */
+    .temp_res = 125,    /**< resolution in 0.125ºC */
+    .temp_mult = 1000,  /**< Must multiply by 1000 to get temp in ºC */
+    .temp_shift = 5,    /**< Only the 11 most significant bits are needed */
+};
+
+/**
+ * @brief params required for initialization
+ */
+typedef struct lm75_params {
+    const lm75_temp_resolutions_t *res;  /**< Temperature resolutions */
+    i2c_t   i2c_bus;                     /**< I2C Bus used */
+    uint8_t shutdown_mode;               /**< Shutdown mode register */
+    uint8_t tm_mode;                     /**< Thermistor Mode */
+    uint8_t polarity;                    /**< OS polarity register */
+    uint8_t fault_q;                     /**< Fault Queue register */
+    uint8_t conv_rate;                   /**< Device Conversion rate - only for the tmp1075 */
+    uint8_t one_shot_mode;               /**< One Shot Mode register - only for the tmp1075 */
+    uint8_t i2c_addr;                    /**< i2c address */
+} lm75_params_t;
+
+/**
+ * @brief lm75 device descriptor
+ */
+typedef struct lm75 {
+    lm75_params_t    lm75_params; /**< Configuration Parameters */
+} lm75_t;
+
+/**
+ * @brief Initialization of the LM75 sensor
+ *
+ * Initializes the sensor according to specific input parameters.
+ *
+ * @param[out] dev        device structure to initialize
+ * @param[in] params      initialization parameters
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on initialization related error
+ */
+int lm75_init(lm75_t *dev, const lm75_params_t *params);
+
+/**
+ * @brief Temperature values of LM75 sensor
+ *
+ * Reads the sensor temperature values from TEMP_REG
+ * the value is given with the full precision the device is capable of
+ *
+ * @param[in] dev                 device structure
+ * @param[in] temperature        buffer where temperature value will be written
```

Comment:
temperature still has no unit :wink:

Reply:
The unit depends on the precision the device is capable of. In the LM75A, for example, it's up to mºC in increments of 125 whereas in the TMP1075 it's up to tenths of mºC in increments of 625.